### PR TITLE
Allow azure upload via a variable

### PR DESCRIPTION
**Pull request by robertpi**
## Summary of changes

Allow artifacts to be stored in Azure bucket by setting a build variable. Useful for end to end testing on branches.

### CODE REVIEW

File: .azure-pipelines/ultimate-pipeline.yml [hunk 1]
```diff
@@ -2889,7 +2889,15 @@ stages:
...
-          condition: and(succeeded(), eq(variables.isMainBranch, true))
+          condition: >
+            and(
+              succeeded(),
+              ne(variables['push_artifacts_to_azure_storage'], 'false'),
+              or(
+                eq(variables['push_artifacts_to_azure_storage'], 'true'),
+                eq(variables.isMainBranch, true)
+              )
+            )
```
Comment (andrewlock) [inline .azure-pipelines/ultimate-pipeline.yml:2900]: The trouble with this is that we use these variables to ensure we're running the latest build in AAS for purposes of the release. Also note that this only uploads the indexes, you're not actually uploading the resources.

Instead, I think you should modify the condition on the Upload blobs to Azure step, and leave this one as it was

**Review by andrewlock — CHANGES_REQUESTED** [review]

File: .azure-pipelines/ultimate-pipeline.yml [hunk 2]
```diff
@@ -2889,7 +2889,15 @@ stages:
             az storage blob upload --container-name "$(AZURE_STORAGE_CONTAINER_NAME)" --file "sha.txt" --name "sha.txt" --overwrite true
             az storage blob upload --container-name "$(AZURE_STORAGE_CONTAINER_NAME)" --file "version.txt" --name "version.txt" --overwrite true
           displayName: Upload indexes to Azure
-          condition: and(succeeded(), eq(variables.isMainBranch, true))
+          condition: >
```
Comment (robertpi) [inline .azure-pipelines/ultimate-pipeline.yml:2892]: This is the only actual change, the rest is just whitespace clean up from VS code. 

I left the whitespace cleanup, as I thought it was probably useful.

File: .azure-pipelines/ultimate-pipeline.yml (hunk 1 above)
Comment (robertpi) [reply .azure-pipelines/ultimate-pipeline.yml]: you're right, thanks, that potentially saved me a lot of confusion about why stuff wasn't be updated!

File: .azure-pipelines/ultimate-pipeline.yml [hunk 3]
```diff
@@ -2870,7 +2870,15 @@ stages:
               --destination-path "$(Build.SourceVersion)" \
               --source "$(Build.ArtifactStagingDirectory)"
           displayName: Upload blobs to Azure
-          condition: and(succeeded(), eq(variables.isMainBranch, true))
+          condition: >
```
Comment (robertpi) [inline .azure-pipelines/ultimate-pipeline.yml:2873]: This is the only actual change, the rest is just whitespace clean up from VS code.

I left the whitespace cleanup, as I thought it was probably useful.

**Review by andrewlock — APPROVED** [review]
LGTM

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #3637 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5): 
* 2 benchmarks are **faster**, with geometric mean 1.235
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`WriteAndFlushEnrichedTraces`|net472|771μs|1.13μs|4.39μs|0.383| 0|0|3.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|551μs|572ns|2.22μs|0| 0|0|2.63&nbsp;KB|
|#3637|`WriteAndFlushEnrichedTraces`|net472|765μs|848ns|3.29μs|0.383| 0|0|3.22&nbsp;KB|
|#3637|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|571μs|526ns|1.97μs|0| 0|0|2.62&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #3637

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.AppSecBodyBenchmark.ObjectExtractorSimpleBody&#8209;net472 | 1.165 | 334.22 | 286.89 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`AllCycleSimpleBody`|net472|23.7μs|35ns|135ns|0.274| 0|0|1.78&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`AllCycleSimpleBody`|netcoreapp3.1|23.9μs|131ns|776ns|0.0213| 0|0|1.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`AllCycleMoreComplexBody`|net472|191μs|108ns|420ns|1.99| 0|0|13.03&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`AllCycleMoreComplexBody`|netcoreapp3.1|180μs|624ns|2.42μs|0.0911| 0|0|12.11&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`ObjectExtractorSimpleBody`|net472|334ns|0.0877ns|0.328ns|0.0573| 0|0|361&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`ObjectExtractorSimpleBody`|netcoreapp3.1|238ns|0.118ns|0.443ns|0.0037| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`ObjectExtractorMoreComplexBody`|net472|15.8μs|11.3ns|43.9ns|1.2| 0.0157|0|7.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`ObjectExtractorMoreComplexBody`|netcoreapp3.1|12.6μs|6.29ns|22.7ns|0.088| 0|0|6.75&nbsp;KB|
|#3637|`AllCycleSimpleBody`|net472|23.1μs|75.5ns|293ns|0.279| 0|0|1.78&nbsp;KB|
|#3637|`AllCycleSimpleBody`|netcoreapp3.1|22.9μs|116ns|505ns|0.0117| 0|0|1.65&nbsp;KB|
|#3637|`AllCycleMoreComplexBody`|net472|194μs|327ns|1.22μs|2.02| 0|0|13.03&nbsp;KB|
|#3637|`AllCycleMoreComplexBody`|netcoreapp3.1|181μs|343ns|1.28μs|0.0906| 0|0|12.11&nbsp;KB|
|#3637|`ObjectExtractorSimpleBody`|net472|287ns|0.301ns|1.17ns|0.0573| 0|0|361&nbsp;B|
|#3637|`ObjectExtractorSimpleBody`|netcoreapp3.1|235ns|0.0808ns|0.313ns|0.00367| 0|0|272&nbsp;B|
|#3637|`ObjectExtractorMoreComplexBody`|net472|15.6μs|10ns|38.8ns|1.21| 0.0156|0|7.62&nbsp;KB|
|#3637|`ObjectExtractorMoreComplexBody`|netcoreapp3.1|12.5μs|4.4ns|17ns|0.088| 0|0|6.75&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #3637

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.AspNetCoreBenchmark.SendRequest&#8209;netcoreapp3.1 | 1.309 | 331,332.26 | 253,022.65 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`SendRequest`|netcoreapp3.1|330μs|1.59μs|6.75μs|0.162| 0|0|20.96&nbsp;KB|
|#3637|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#3637|`SendRequest`|netcoreapp3.1|252μs|1.37μs|7.48μs|0.253| 0|0|20.96&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`ExecuteNonQuery`|net472|1.88μs|2.65ns|9.91ns|0.162| 0.000933|0|1.03&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`ExecuteNonQuery`|netcoreapp3.1|1.45μs|2.39ns|9.27ns|0.0137| 0|0|1.02&nbsp;KB|
|#3637|`ExecuteNonQuery`|net472|1.84μs|3.16ns|11.8ns|0.163| 0.000921|0|1.03&nbsp;KB|
|#3637|`ExecuteNonQuery`|netcoreapp3.1|1.54μs|1.48ns|5.72ns|0.0139| 0|0|1.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`CallElasticsearch`|net472|2.44μs|1.14ns|4.4ns|0.196| 0|0|1.24&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`CallElasticsearch`|netcoreapp3.1|1.53μs|1.16ns|4.17ns|0.0155| 0|0|1.18&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`CallElasticsearchAsync`|net472|2.57μs|1.56ns|6.05ns|0.217| 0|0|1.37&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`CallElasticsearchAsync`|netcoreapp3.1|1.61μs|4.06ns|15.2ns|0.0176| 0|0|1.3&nbsp;KB|
|#3637|`CallElasticsearch`|net472|2.51μs|0.526ns|2.04ns|0.195| 0|0|1.24&nbsp;KB|
|#3637|`CallElasticsearch`|netcoreapp3.1|1.48μs|0.655ns|2.45ns|0.0156| 0|0|1.18&nbsp;KB|
|#3637|`CallElasticsearchAsync`|net472|2.62μs|0.68ns|2.54ns|0.217| 0|0|1.37&nbsp;KB|
|#3637|`CallElasticsearchAsync`|netcoreapp3.1|1.62μs|0.474ns|1.77ns|0.0179| 0|0|1.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`ExecuteAsync`|net472|2.69μs|1.3ns|5.05ns|0.238| 0|0|1.5&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`ExecuteAsync`|netcoreapp3.1|1.7μs|0.527ns|1.83ns|0.0188| 0|0|1.42&nbsp;KB|
|#3637|`ExecuteAsync`|net472|2.67μs|0.971ns|3.76ns|0.238| 0|0|1.5&nbsp;KB|
|#3637|`ExecuteAsync`|netcoreapp3.1|1.77μs|0.917ns|3.55ns|0.0194| 0|0|1.42&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`SendAsync`|net472|7.16μs|1.75ns|6.55ns|0.556| 0|0|3.51&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`SendAsync`|netcoreapp3.1|4.59μs|1.5ns|5.82ns|0.0438| 0|0|3.21&nbsp;KB|
|#3637|`SendAsync`|net472|7.17μs|2.74ns|10.2ns|0.559| 0|0|3.51&nbsp;KB|
|#3637|`SendAsync`|netcoreapp3.1|4.69μs|3.85ns|14.4ns|0.0421| 0|0|3.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`EnrichedLog`|net472|2.71μs|1.12ns|3.87ns|0.302| 0|0|1.9&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`EnrichedLog`|netcoreapp3.1|2.26μs|0.824ns|3.08ns|0.0261| 0|0|1.93&nbsp;KB|
|#3637|`EnrichedLog`|net472|2.68μs|0.963ns|3.47ns|0.302| 0|0|1.9&nbsp;KB|
|#3637|`EnrichedLog`|netcoreapp3.1|2.23μs|0.855ns|3.08ns|0.0268| 0|0|1.93&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`EnrichedLog`|net472|147μs|180ns|697ns|0.734| 0.22|0|4.74&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`EnrichedLog`|netcoreapp3.1|117μs|106ns|397ns|0.0582| 0|0|4.58&nbsp;KB|
|#3637|`EnrichedLog`|net472|147μs|67.8ns|263ns|0.733| 0.22|0|4.74&nbsp;KB|
|#3637|`EnrichedLog`|netcoreapp3.1|116μs|204ns|790ns|0.058| 0|0|4.58&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`EnrichedLog`|net472|5.64μs|3.02ns|11.7ns|0.582| 0.0028|0|3.68&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`EnrichedLog`|netcoreapp3.1|4.23μs|1.63ns|6.1ns|0.0549| 0|0|4&nbsp;KB|
|#3637|`EnrichedLog`|net472|5.49μs|0.817ns|2.95ns|0.583| 0.00275|0|3.68&nbsp;KB|
|#3637|`EnrichedLog`|netcoreapp3.1|4.22μs|1.74ns|6.5ns|0.0527| 0|0|4&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`SendReceive`|net472|2.24μs|1.09ns|4.22ns|0.23| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`SendReceive`|netcoreapp3.1|1.73μs|1.17ns|4.53ns|0.0191| 0|0|1.4&nbsp;KB|
|#3637|`SendReceive`|net472|2.1μs|1.12ns|4.33ns|0.23| 0|0|1.45&nbsp;KB|
|#3637|`SendReceive`|netcoreapp3.1|1.67μs|0.651ns|2.44ns|0.0192| 0|0|1.4&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`EnrichedLog`|net472|4.69μs|1.81ns|7ns|0.368| 0|0|2.32&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`EnrichedLog`|netcoreapp3.1|3.99μs|4.24ns|16.4ns|0.0259| 0|0|1.89&nbsp;KB|
|#3637|`EnrichedLog`|net472|4.6μs|1.78ns|6.41ns|0.367| 0|0|2.32&nbsp;KB|
|#3637|`EnrichedLog`|netcoreapp3.1|3.98μs|2.06ns|7.69ns|0.0258| 0|0|1.89&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`StartFinishSpan`|net472|1.12μs|0.36ns|1.35ns|0.141| 0|0|891&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`StartFinishSpan`|netcoreapp3.1|938ns|0.276ns|1.03ns|0.0112| 0|0|840&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`StartFinishScope`|net472|1.39μs|0.504ns|1.88ns|0.154| 0|0|971&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`StartFinishScope`|netcoreapp3.1|1.01μs|0.42ns|1.63ns|0.0131| 0|0|960&nbsp;B|
|#3637|`StartFinishSpan`|net472|1.11μs|0.377ns|1.41ns|0.141| 0|0|891&nbsp;B|
|#3637|`StartFinishSpan`|netcoreapp3.1|936ns|0.434ns|1.62ns|0.0111| 0|0|840&nbsp;B|
|#3637|`StartFinishScope`|net472|1.35μs|0.401ns|1.5ns|0.154| 0|0|971&nbsp;B|
|#3637|`StartFinishScope`|netcoreapp3.1|1.05μs|0.304ns|1.14ns|0.0132| 0|0|960&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`RunOnMethodBegin`|net472|1.55μs|0.527ns|1.9ns|0.154| 0|0|971&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f0a692631c4b9244482bb047f280e8fd87aa62f5)|`RunOnMethodBegin`|netcoreapp3.1|1.16μs|0.345ns|1.34ns|0.0128| 0|0|960&nbsp;B|
|#3637|`RunOnMethodBegin`|net472|1.43μs|0.354ns|1.28ns|0.154| 0|0|971&nbsp;B|
|#3637|`RunOnMethodBegin`|netcoreapp3.1|1.09μs|0.315ns|1.09ns|0.0131| 0|0|960&nbsp;B|
</details>

File: .azure-pipelines/ultimate-pipeline.yml (hunk 3 above)
Comment (zacharycmontoya) [reply .azure-pipelines/ultimate-pipeline.yml:2873]: GitHub PR > ⚙️ > `Hide whitespace` 👌🏼

File: .azure-pipelines/ultimate-pipeline.yml [hunk 4]
```diff
@@ -2870,7 +2870,15 @@ stages:
...
+            and(
+              succeeded(),
+              ne(variables['push_artifacts_to_azure_storage'], 'false'),
```
Comment (zacharycmontoya) [inline .azure-pipelines/ultimate-pipeline.yml:2876]: Potentially a silly question, but is this line needed? I suppose it adds the ability to disable Azure pushes if one sets `push_artifacts_to_azure_storage == false`, but I haven't run into that before

**Review by zacharycmontoya — APPROVED** [review]

File: .azure-pipelines/ultimate-pipeline.yml (hunk 4 above)
Comment (andrewlock) [reply .azure-pipelines/ultimate-pipeline.yml:2876]: Yeah, agreed, I can't see a need for this line🤔

File: .azure-pipelines/ultimate-pipeline.yml (hunk 3 above)
Comment (robertpi) [reply .azure-pipelines/ultimate-pipeline.yml:2873]: Wow, didn't know about that, thanks!

File: .azure-pipelines/ultimate-pipeline.yml (hunk 4 above)
Comment (robertpi) [reply .azure-pipelines/ultimate-pipeline.yml:2876]: I copy / pasted it from the s3 upload, which includes the following justification:
https://github.com/DataDog/dd-trace-dotnet/pull/3637/files#diff-2e09bd52f55cbb3965ceeae7f556dbbd458b94a4566d9631bea1204a4ad12198R2686

The false allows an explicit block, for the master branch. Might be useful at some point, but don't have strong opinion.

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #3637 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/628f4d00fea055949660c271d768d0f7ece1030b) will **not change** line coverage
:heavy_check_mark: Merging #3637 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/628f4d00fea055949660c271d768d0f7ece1030b) will **not change** branch coverage
:heavy_check_mark: Merging #3637 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/628f4d00fea055949660c271d768d0f7ece1030b) will will **decrease** complexity by `133`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/628f4d00fea055949660c271d768d0f7ece1030b) | #3637       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `21080` / `29717` | `21072` / `29529` |          |
| Lines %   | `71%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `12666` / `19155` | `12634` / `19035` |          |
| Branches %| `66%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `20667`      | `20534`        |  `-133`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=124540&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/124540/artifacts?artifactName=Code%20Coverage%20Report_124540&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3637](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=124278&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/124278/artifacts?artifactName=Code%20Coverage%20Report_124278&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/628f4d00fea055949660c271d768d0f7ece1030b) | #3637       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `66%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `20667`      | `20534`       |  `-133` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestClassInfoRunClassCleanupIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/e5c2a9ea7f3cbd93b890da43532522eb6dc705de/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestClassInfoRunClassCleanupIntegration.cs) | `-70%` :no_entry: | `-58%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.Payloads.EventPlatformPayload](https://github.com/DataDog/dd-trace-dotnet/tree/e5c2a9ea7f3cbd93b890da43532522eb6dc705de/tracer/src/Datadog.Trace/Ci/Agent/Payloads/EventPlatformPayload.cs) | `4%` :heavy_check_mark: | `12%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/e5c2a9ea7f3cbd93b890da43532522eb6dc705de/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `6%` :heavy_check_mark: | `-2%` :warning:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Util.AsyncUtil](https://github.com/DataDog/dd-trace-dotnet/tree/e5c2a9ea7f3cbd93b890da43532522eb6dc705de/tracer/src/Datadog.Trace/Util/AsyncUtil.cs) | `19%` :heavy_check_mark: | `100%` :heavy_check_mark:   | `-6` :heavy_check_mark: |


5 classes were removed from Datadog.Trace in #3637


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=124540&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/124540/artifacts?artifactName=Code%20Coverage%20Report_124540&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3637](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=124278&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/124278/artifacts?artifactName=Code%20Coverage%20Report_124278&api-version=6.0&%24format=zip)

File: .azure-pipelines/ultimate-pipeline.yml (hunk 3 above)
Comment (andrewlock) [reply .azure-pipelines/ultimate-pipeline.yml:2873]: > This is github desktop only?

Web view 🙂 

image

**Pull request merged**